src/go-slang/parser/go.pegjs

```diff
@@ -185,6 +189,46 @@ CallExpression
         return makeNode({ type: "CallExpression", callee, args: args ?? [] })
       }
 
+/* Select Statement */
+
+SelectStatement "select statement"
+    = SELECT_TOKEN _ "{" _ cases:CommClause* _ "}" EOS {
+        return { type: "SelectStatement", cases }
```

Comment:
Also, we can wrap every node creation with `makeNode` to inject UID and location information which will come in handy for ECE implementation later on

Reply:
Done.